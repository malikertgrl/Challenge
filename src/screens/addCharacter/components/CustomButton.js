import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import React from 'react'
import { Colors } from "../../../constants"

const CustomButton = ({ onPress }) => {
    return (
        <TouchableOpacity onPress={onPress}>
            <View style={styles.button}>
                <Text style={styles.text}>Add Character</Text>
            </View>
        </TouchableOpacity>

    )
}

export default CustomButton

const styles = StyleSheet.create({
    button: {
        backgroundColor: Colors.blue,
        height: 45,
        margin: 10,
        alignItems: "center",
        justifyContent: "center",
        borderRadius: 5
    },
    text: {
        color: Colors.white,
    }
})