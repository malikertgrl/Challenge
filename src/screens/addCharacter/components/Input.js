import { StyleSheet, Text, View, TextInput } from 'react-native'
import React from 'react'
import { Colors } from "../../../constants"
const Input = ({
    testID,
    onChangeText,
    value,
    title,
    placeholder,
    style }) => {
    return (
        <View style={{ margin: 10 }}>
            <Text style={styles.text}>{title}</Text>
            <TextInput
                testID={testID}
                style={[styles.inputStyle, { ...style }]}
                value={value}
                onChangeText={onChangeText}
                placeholder={placeholder}
                multiline={true}
            />
        </View>
    )
}

export default Input

const styles = StyleSheet.create({
    inputStyle: {
        backgroundColor: "#fff",
        marginVertical: 5,
        elevation: 2,
        height: 45,
        borderRadius: 5
    },
    text: { color: Colors.black }
})