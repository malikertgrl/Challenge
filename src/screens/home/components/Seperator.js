import { StyleSheet, Text, View } from 'react-native'
import React from 'react'

const Seperator = () => {
    return (
        <View style={styles.SeperatorStyle}>
        </View>
    )
}

export default Seperator

const styles = StyleSheet.create({
    SeperatorStyle: {
        borderBottomWidth: 1,
        borderBottomColor: "#0001"
    }
})