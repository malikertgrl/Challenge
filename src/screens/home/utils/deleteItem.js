export const deleteItem = (id, data) => {
    const newData = data.filter(item => item.id != id)
    return newData;
}