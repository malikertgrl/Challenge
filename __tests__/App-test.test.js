import React from "react";
import renderer from 'react-test-renderer';
import { render, fireEvent } from "@testing-library/react-native"
import Seperator from "../src/screens/home/components/Seperator"
import CustomButton from "../src/screens/AddCharacter/components/CustomButton"
import Input from "../src/screens/AddCharacter/components/Input"
import AddCharacter from "../src/screens/addCharacter"
import { deleteItem } from "../src/screens/home/utils/deleteItem"


test('renders seperator', () => {
  const tree = renderer.create(<Seperator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders customButton', () => {
  const tree = renderer.create(<CustomButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Input', () => {
  const tree = renderer.create(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});


describe("Controlling text inputs", () => {

  it("Checking text input with testID = name", () => {
    const setStateMock = jest.fn();
    const useStateMock = (useState) => [useState, setStateMock];
    jest.spyOn(React, "useState").mockImplementation(useStateMock); //reactaki usestate yerine kullan覺caz

    const { getByTestId } = render(<AddCharacter />);
    const Input = getByTestId("name");
    fireEvent.changeText(Input, 'test')
    expect(getByTestId('name').props.value).toEqual('test');


  })

  it("Checking text input with testID = job", () => {
    const setStateMock = jest.fn();
    const useStateMock = (useState) => [useState, setStateMock];
    jest.spyOn(React, "useState").mockImplementation(useStateMock); //reactaki usestate yerine kullan覺caz

    const { getByTestId } = render(<AddCharacter />);
    const Input = getByTestId("job");
    fireEvent.changeText(Input, 'test')
    expect(getByTestId('job').props.value).toEqual('test');


  })
  it("Checking text input with testID = about", () => {
    const setStateMock = jest.fn();
    const useStateMock = (useState) => [useState, setStateMock];
    jest.spyOn(React, "useState").mockImplementation(useStateMock); //reactaki usestate yerine kullan覺caz

    const { getByTestId } = render(<AddCharacter />);
    const Input = getByTestId("about");
    fireEvent.changeText(Input, 'test')
    expect(getByTestId('about').props.value).toEqual('test');


  })

  it("Checking text input with testID = avatar", () => {
    const setStateMock = jest.fn();
    const useStateMock = (useState) => [useState, setStateMock];
    jest.spyOn(React, "useState").mockImplementation(useStateMock); //reactaki usestate yerine kullan覺caz

    const { getByTestId } = render(<AddCharacter />);
    const Input = getByTestId("avatar");
    fireEvent.changeText(Input, 'test')
    expect(getByTestId('avatar').props.value).toEqual('test');


  })

})

it("delete item", () => {
  const data = [
    { name: "malik", job: "developer", about: "blabla", id: "1" },
    { name: "Soner", job: "developer", about: "blabla", id: "2" }
  ]

  expect(deleteItem(1, data)).toHaveLength(1)
})




