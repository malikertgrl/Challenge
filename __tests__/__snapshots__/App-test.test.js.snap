// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#0001",
      "borderBottomWidth": 1,
    }
  }
/>
`;

exports[`renders correctly 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1e88e5",
        "borderRadius": 5,
        "height": 45,
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      Add Character
    </Text>
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
      }
    }
  />
  <TextInput
    multiline={true}
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "elevation": 2,
          "height": 45,
          "marginVertical": 5,
        },
        Object {},
      ]
    }
  />
</View>
`;
